api/proto/external.proto

```diff
@@ -255,7 +255,10 @@ message RingMLSAG {
 message SignatureRctBulletproofs {
     repeated RingMLSAG ring_signatures = 1;
     repeated CompressedRistretto pseudo_output_commitments = 2;
-    bytes range_proofs = 3;
+    bytes range_proof_bytes = 3;
```

Comment:
This should maybe be a oneof (range_proof_bytes/range_proofs) since you only want to be able to have either the range_proof_bytes or range_proofs.

Reply:
Per discussion, adding a oneof in this case is likely backwards compatible in protobuf, however we would likely encounter difficulties due to potential conflicts in the hash of old tx structure in the mc-crypt-digestible crate. With additive changes we can avoid this conflict because new fields simply hash to 0.

Reply:
>  simply hash to 0.

simply don't contribute to the hash when they are `None`

yeah...

we could add a OneOf with one option, but I think what we actually want to do here is like, make `SignatureRctBulletproofsv2` or something and just start over with this structure, and do schema evolution in `Tx` instead